src/pint/models/wavex.py

```diff
@@ -0,0 +1,387 @@
+    of harmonics of a base frequency determined by 2 * pi /Timespan
+    """
+
+    register = True
+    category = "wavex"
+
+    def __init__(self):
+        super().__init__()
+        self.add_param(
+            MJDParameter(
+                name="WXEPOCH",
+                description="Reference epoch for wave delay solution",
+                time_scale="tdb",
+            )
+        )
+        self.add_wavex_component(0.1, index=1, wxsin=0, wxcos=0, frozen=False)
+        self.set_special_params(["WXFREQ_0001", "WXSIN_0001", "WXCOS_0001"])
+        self.delay_funcs_component += [self.wavex_delay]
+
+    def add_wavex_component(self, wxfreq, index=None, wxsin=0, wxcos=0, frozen=True):
+        """
+        Add WaveX component
+
+        Parameters
+        ----------
+
+        wxfreq : float or astropy.quantity.Quantity
+            Base frequency for WaveX component
+        index : int, None
+            Interger label for WaveX component. If None, will increment largest used index by 1.
+        wxsin : float or astropy.quantity.Quantity
+            Sine amplitude for WaveX component
+        wxcos : float or astropy.quantity.Quantity
+            Cosine amplitude for WaveX component
+        frozen : iterable of bool or bool
+            Indicates whether wavex will be fit
+
+        Returns
+        -------
+
+        index : int
+            Index that has been assigned to new WaveX component
+        """
+
+        #### If index is None, increment the current max WaveX index by 1. Increment using WXFREQ
+        if index is None:
+            dct = self.get_prefix_mapping_component("WXFREQ_")
+            index = np.max(list(dct.keys())) + 1
+        i = f"{int(index):04d}"
+
+        if int(index) in self.get_prefix_mapping_component("WXFREQ_"):
+            raise ValueError(
+                f"Index '{index}' is already in use in this model. Please choose another"
+            )
+
+        if isinstance(wxsin, u.quantity.Quantity):
+            wxsin = wxsin.to_value(u.s)
+        if isinstance(wxcos, u.quantity.Quantity):
+            wxcos = wxcos.to_value(u.s)
+        if isinstance(wxfreq, u.quantity.Quantity):
+            wxfreq = wxfreq.to_value(1 / u.d)
+        self.add_param(
+            prefixParameter(
+                name=f"WXFREQ_{i}",
+                description="Base frequency of wave delay solution",
+                units="1/d",
+                value=wxfreq,
+                parameter_type="float",
+            )
+        )
+        self.add_param(
+            prefixParameter(
+                name=f"WXSIN_{i}",
+                description="Sine amplitudes for wave delay function",
+                units="s",
+                value=wxsin,
+                frozen=frozen,
+                parameter_type="float",
+            )
+        )
+        self.add_param(
+            prefixParameter(
+                name=f"WXCOS_{i}",
+                description="Cosine amplitudes for wave delay function",
```

Comment:
Cos amplitude for Fourier representation of red noise {i}

Reply:
Changed